pivotx/modules/module_parser.php

```diff
@@ -2579,7 +2594,8 @@ function cms_tag_weblog($params, $format){
     //}
     
     // Restore the old entry..
-    $PIVOTX['db']->set_entry($old_entry);
+    if( $old_entry != null) // Testing early allows to avoid triggering unnecessary index udpate (i.e. heavy db file writing)
```

Comment:
Not directly linked to db corruption, but optimizing unnecessary db index update reduces issue linked to concurrent access.